### PR TITLE
ENH: Implement a drop-in replacement for RunEngine Document Dispatcher.

### PR DESCRIPTION
**Pull request by danielballan**
We documented this but never actually wrote it.

Having trouble running tests locally due to pip problems + pymongo thrashing.

### CODE REVIEW

File: databroker/databroker.py [hunk 1]
```diff
@@ -423,3 +423,45 @@ def get_table(headers, fields=None, fill=True, convert_times=True):
...
+    headers : Header or iterable of Headers
+        The headers to fetch the events for
+    fields : list, optional
+        whitelist of field names of interest; if None, all are returned
+    fill : bool, optional
+        Whether externally-stored data should be filled in. Defaults to True
+
+    Yields
+    ------
+    doc : Document
```
Comment (tacaswell) [inline databroker/databroker.py]: I think this should `yield name, doc` as that is the signature that the call backs expect.

Comment (danielballan) [inline databroker/databroker.py]: Oh, duh. Fully agree!

File: databroker/tests/test_broker.py [hunk 2]
```diff
@@ -242,3 +242,22 @@ def test_raise_conditions():
     ]
     for raiser in raising_keys:
         yield _raiser_helper, raiser
+
+
+def test_stream():
+    s = stream(db[-1])
```
Comment (tacaswell) [inline databroker/tests/test_broker.py]: This isn't safe for testing.  We really need to sort out py.test paramaterized tests.

Comment (ericdill) [inline databroker/tests/test_broker.py]: There is a module level `setup()` function that gets run first, so the broker will always have access to a header

Comment (tacaswell) [inline databroker/tests/test_broker.py]: This assumes a one descriptor event stream

Comment (danielballan) [inline databroker/tests/test_broker.py]: > We really need to sort out py.test paramaterized tests.

... in this PR?

Comment (tacaswell) [inline databroker/tests/test_broker.py]: Or just insert a data stream and retrieve it for this test.

**Pull request merged**